--- a/cmake/LAPACKConfig.cmake
+++ b/cmake/LAPACKConfig.cmake
@@ -73,6 +73,14 @@
 #-------------------------------------------------------------------------------
 # Find LAPACKE, either in the BLAS/LAPACK library or in -llapacke.
 # Check for pstrf (Cholesky with pivoting).
+
+# For libraries in our prefix, add include and library paths so try_run can find lapacke.
+if (DEFINED CMAKE_INCLUDE_PATH)
+    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -I${CMAKE_INCLUDE_PATH}" )
+endif()
+if (DEFINED CMAKE_LIBRARY_PATH)
+    set( CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -L${CMAKE_LIBRARY_PATH}" )
+endif()

 set( lib_list ";-llapacke" )
 message( DEBUG "lib_list ${lib_list}" )
