--- a/examples/cvode/petsc/cv_petsc_ex7.c
+++ b/examples/cvode/petsc/cv_petsc_ex7.c
@@ -154,7 +154,7 @@ int main(int argc, char** argv)
                         (PetscErrorCode(*)(SNES, PetscInt, PetscReal,
                                            void*))MySNESMonitor,
                         vf,
-                        (PetscErrorCode(*)(void**))PetscViewerAndFormatDestroy);
+                        (PetscErrorCode(*)(void*))PetscViewerAndFormatDestroy);
   CHKERRQ(ierr);

   /* use matrix free */
